onnxruntime/contrib_ops/rocm/bert/tunable_op.h

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#pragma once
+
+#include <iostream>
+#include <map>
+#include <vector>
+#include <string>
+#include "timer.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace rocm {
+
+struct OpParams {
+  virtual std::string signature() const = 0;
+};
+
+class Op {
+ public:
+  Op() : repeats_(100) {}
+
+  virtual void Run(const OpParams*) = 0;
+
+  void SetRepeats(int n) {
+    repeats_ = n;
+  }
+
+  float Profile(const OpParams* op_params) {
```

Comment:
besides the performance, do we want to verify the correctness? in case if any configuration cause hidden bug in the kernrels.

Reply:
Yes, we do need better testing coverage for correctness for the various template instantiations; I'm thinking about add these with python testing frontend.